tests/console/commands/test_check.py

```diff
@@ -111,19 +127,55 @@ def test_check_valid_legacy(
     assert tester.io.fetch_error() == expected
 
 
-def test_check_invalid(
+def test_check_invalid_dep_name_same_as_project_name(
```

Comment:
What is the reason for splitting (almost duplicating) the test and how is it related to the fix?

Reply:
The old one was checking the dependency name vs project name. This is not possible with a fixture being a Poetry instance, so I moved this to another test to keep this case tested.

Reply:
So if we do not introduce the fixture, we do not have to split the test? Then, why introduce a fixture in the first place? (Further, it still seems unrelated to the test case for the fix.)

Reply:
We need to introduce the fixture because otherwise (with patching pyproject file) we get the wrong list of sources for the project.

Reply:
I still do not get why this is an issue. We now have two almost similar tests (`test_check_invalid_dep_name_same_as_project_name `and `test_check_invalid`). Both provide the same errors except for that the first one provides two more errors:

- `Error: Project name (invalid) is same as one of its dependencies`
- `Error: Invalid source "exists" referenced in dependencies.`

And both tests are unrelated to the issue that is resolved by this PR. This is tested in `test_check_does_not_error_on_pypi_reference`, which uses another fixture.

Reply:
After trying it out myself, I now understand why you had to introduce the fixture: The change requires that sources are parsed, which is not done without the fixture. Otherwise, all sources (also existing ones) will be recognized as non-existent.

I added a commit to reduce the duplication between the two tests to a minimum. Please take a look if this is fine for you.

Reply:
Looks fine, thanks for the commit.